src/note.rs

```diff
@@ -226,6 +227,25 @@ impl Note {
         }
     }
 
+    /// Reconstruct a Note from its individual fields.
+    pub fn reconstruct(
```

Comment:
IDK if the name is really telling what we're doing here. I think methods like `new` or `from_fields` would fit better.

Reply:
Fair enough. `new` is already taken so I was unsure of what name to pick - maybe `from_fields` is better.